conformance/tests/basic/inferencepool_no_matching_pods_route_status.go

```diff
@@ -0,0 +1,89 @@
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package basic
+
+import (
+	"fmt"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	gatewayv1 "sigs.k8s.io/gateway-api/apis/v1"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+	"sigs.k8s.io/gateway-api/pkg/features"
+
+	"sigs.k8s.io/gateway-api-inference-extension/conformance/tests"
+	infrakubernetes "sigs.k8s.io/gateway-api-inference-extension/conformance/utils/kubernetes"
+)
+
+func init() {
+	tests.ConformanceTests = append(tests.ConformanceTests, InferencePoolNoMatchingPodsRouteStatus)
+}
+
+var InferencePoolNoMatchingPodsRouteStatus = suite.ConformanceTest{
+	// TODO: Update based on the outcome of https://github.com/kubernetes-sigs/gateway-api-inference-extension/issues/806
+
+	ShortName:   "InferencePoolNoMatchingPodsRouteStatus",
+	Description: "Tests HTTPRoute and Gateway status when an HTTPRoute references an InferencePool whose modelServerSelector does not match any running pods.",
+	Manifests:   []string{"tests/basic/inferencepool_no_matching_pods_route_status.yaml"},
+	Features:    []features.FeatureName{},
+	Test: func(t *testing.T, s *suite.ConformanceTestSuite) {
+		const (
+			appBackendNamespace   = "gateway-conformance-app-backend"
+			infraNamespace        = "gateway-conformance-infra"
+			poolName              = "pool-no-pods"
+			httpRouteName         = "httproute-for-pool-no-pods"
+			gatewayName           = "conformance-gateway"
+			expectedRouteReason   = "ReconciliationFailed"
+			expectedGatewayReason = "Invalid"
+		)
+
+		poolNN := types.NamespacedName{Name: poolName, Namespace: appBackendNamespace}
+		routeNN := types.NamespacedName{Name: httpRouteName, Namespace: appBackendNamespace}
+		gatewayNN := types.NamespacedName{Name: gatewayName, Namespace: infraNamespace}
+
+		t.Logf("Verifying InferencePool %s is initially accepted by Gateway %s (via HTTPRoute %s)", poolNN.String(), gatewayNN.String(), routeNN.String())
+		acceptedCondition := metav1.Condition{
+			Type:   string(gatewayv1.RouteConditionAccepted),
+			Status: metav1.ConditionTrue,
+			Reason: string(gatewayv1.RouteReasonAccepted),
+		}
+		infrakubernetes.InferencePoolMustHaveCondition(t, s.Client, poolNN, acceptedCondition)
+		t.Logf("InferencePool %s parent status shows Accepted by Gateway %s", poolNN.String(), gatewayNN.String())
+
+		t.Logf("Polling for HTTPRoute %s status to reflect backend issues...", routeNN.String())
+
+		expectedAcceptedRouteCond := metav1.Condition{
+			Type:   string(gatewayv1.RouteConditionAccepted),
+			Status: metav1.ConditionTrue,
+		}
+
+		expectedFailureRouteCond := metav1.Condition{
+			Type:   string(gatewayv1.RouteConditionType("Reconciled")),
```

Comment:
"Reconciled" is not a condition type defined by Gateway API.

Reply:
That is a great point! You're absolutely right that `Reconciled` isn't a formally defined `gatewayv1.RouteConditionType`. I've observed that the controller under test does emit a status condition with Type: "Reconciled" (with Reason: "ReconciliationSucceeded" on success, or ReconciliationFailed in cases like this test). This test is currently asserting against that observed behavior.  

To clarify this, I've added a code comment with a TODO and will open a new issue #846 to discuss whether this should be formally defined in Gateway API or if another existing condition should be leveraged. I'll link the issue here for broader visibility and discussion.